core/views.py

```diff
@@ -1,22 +1,113 @@
-from rest_framework import permissions
-from rest_framework import generics
-from rest_framework import status
-from .serializers import HackathonSerializer
-from .models import Hackathon
-from .permissions import HackathonPermissions
-
-class HackathonCreateView(generics.CreateAPIView):
-    '''
-    API used to create hackathon objects. Can only be accessed by the Super User
-    '''
+from rest_framework import generics, status, permissions, exceptions
+from rest_framework.response import Response
+from django.utils import timezone
+from django.utils.decorators import method_decorator
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
+from .models import Hackathon, Team
+from .serializers import HackathonSerializer, TeamSerializer, TeamCreateSerializer, JoinTeamSerializer
+from .permissions import HackathonPermissions, AllowCompleteProfile
+
+class HackathonTeamView(generics.ListCreateAPIView):
+    """
+    get:
+    Returns a list of teams in a particular hackathon
+    post:
+    Creates a new team in a hackathon and return the team_id
+    """
+    permission_classes = [permissions.IsAuthenticated, AllowCompleteProfile]
+
+    def get_serializer_class(self):
+        if self.request.method == 'GET':
+            return TeamSerializer
+        else:
+            return TeamCreateSerializer
+    def get_serializer_context(self):
+        return {
+            'request': self.request,
+            'kwargs': self.kwargs
+        }
+
+    def get_queryset(self, **kwargs):
+        if getattr(self, 'swagger_fake_view', False):
+            return None
+        try:
+            hackathon = Hackathon.objects.get(id=self.kwargs['pk'])
+        except Hackathon.DoesNotExist:
+            raise exceptions.NotFound("Hackathon does not exist!")
+        queryset = Team.objects.filter(hackathon=self.kwargs['pk'])
+        return queryset
+
+    def post(self, request, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        team = serializer.save()
+        serializer = TeamSerializer(team)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+class JoinTeamView(generics.GenericAPIView):
+    """
+    patch:
+    Join a team with team_id and hackathon_id
+    """
+    serializer_class = JoinTeamSerializer
+
+    def get_queryset(self):
+        if getattr(self, 'swagger_fake_view', False):
+            return None
+        pass
```

Comment:
```suggestion
        return super().get_object()
```
Shouldn't we call the parent class function here?
/cc @nishantwrp